### PR TITLE
Adding rst API docs

### PR DESCRIPTION
**Pull request by essamROCm**
Adding all the RST files for the rocPyDecode API

### CODE REVIEW

File: docs/reference/rocPyDecode.rst [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+**********************
+rocPyDecode Python API
+**********************
+
+The rocPyDecode API is a ROCm rocDecode Python/C++ Binding, a tool that lets you access rocDecode APIs in both Python and C/C++ languages. It works by connecting Python and C/C++ libraries, enabling function calling and data passing between the two languages. The ``rocpydecode`` library is a wrapper API that uses the rocDecode C/C++ language APIs within a Python wrapper.
+
+.. image:: ../data/reference/rocPyDecodeArch.png
+
+The rocPyDecode API main classes are a decoder class and a demuxer class. All rocPyDecode APIs are exposed using the header files ``decoder.py`` and ``demuxer.py``. You can find these files in the `pyRocVideoDecode <https://github.com/ROCm/rocPyDecode/blob/develop/pyRocVideoDecode>`_ folder in the rocPyDecode github repository.
```
Comment (randyh62) [inline docs/reference/rocPyDecode.rst]: The decoder and demuxer classes seem related to the ``PyRocVideoDecoder`` and ``PyVideoDemuxer`` classes discussed in the _Using rocPyDecode_ topic. If these are the same then we should sort out the naming of these classes. It looks like the functions are similar, though there are some differences.

**Review by randyh62 — APPROVED** [review]

File: docs/reference/rocPyDecode.rst (hunk 1 above)
Comment (essamROCm) [reply docs/reference/rocPyDecode.rst]: The decoder class in the decoder.py is python class, which is bind to the C++ class PyRocVideoDecoder, and so the demuxer class in the demuxer.py is a python class, which is bind to the C++ class PyVideoDemuxer. The naming was by design to distinguish among the 2 different spaces.

Comment (randyh62) [reply docs/reference/rocPyDecode.rst]: Maybe worth adding to either the Using rocPyDecode topic or Reference topic?

File: docs/data/reference/rocPyDecodeArch.png
Comment (rrawther) [inline docs/data/reference/rocPyDecodeArch.png:1]: I think we need to change the color of the arrows for this picture as it is not visible in the white background. Also I feel the perforated background window is not looking good as well

File: docs/reference/deCoderClass.rst
Comment (rrawther) [inline docs/reference/deCoderClass.rst:1]: typo "deCoderClass" please change to decoderClass.rst

File: docs/reference/deCoderClass.rst [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+Decodes video frames described by the input :ref:`packet` obtained from demux functions. Returns the count of decoded frames.
+
+
+GetFrame(packet)
+----------------
+
+Obtains a pointer to the current decoded frame in the input packet. Returns the current frame time stamp, the frame pointer, and the time stamp saved in the packet.
+
+Example:
+^^^^^^^^
```
Comment (rrawther) [inline docs/reference/deCoderClass.rst]: please be consistent in underlining format. previous is "----"

File: docs/reference/deMuxerClass.rst
Comment (rrawther) [inline docs/reference/deMuxerClass.rst:1]: same here. demuxerClass.rst is correct name

File: docs/reference/deMuxerClass.rst [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+- **seek_criteria**:  0 - Seek by frame number, 1 - Seek by time stamp
+
+Example:
+^^^^^^^^
+
+.. code-block:: python
+	
+		not_seeking = False
+
+		seek_frame = 100     # seek 100 frames
```
Comment (rrawther) [inline docs/reference/deMuxerClass.rst]: change comment to "seek to 100th frame"

**Review by rrawther — CHANGES_REQUESTED** [review]
added review comments

File: docs/reference/deCoderClass.rst (hunk 2 above)
Comment (randyh62) [reply docs/reference/deCoderClass.rst]: These are header levels: 
===== is H1, ----- is H2, and ^^^^ is H3 level headers

File: docs/reference/rocPyDecode.rst (hunk 1 above)
Comment (essamROCm) [reply docs/reference/rocPyDecode.rst]: > Maybe worth adding to either the Using rocPyDecode topic or Reference topic?

Can you please explain this request in teams? I cannot get it from the question here.

Comment (essamROCm) [reply docs/reference/rocPyDecode.rst]: Per our conversation, I added more explanation to clarify [here in this last commit](https://github.com/ROCm/rocPyDecode/pull/80/commits/73c55c95eb060ee05fd24c276e53d8e3a935144e).

**Review by kiritigowda — APPROVED** [review]

**Pull request merged**